karate-core2/src/test/java/com/intuit/karate/match/MatchTest.java

```diff
@@ -239,6 +239,23 @@ void testXml() {
     void testXmlSchema() {
         match("<root></root>", EQUALS, "<root>#null</root>"); // TODO controversial
         match("<root></root>", EQUALS, "<root>#present</root>");
+        match("<root><a>x</a><b><c>y</c></b></root>", EQUALS, "<root><a>#string</a><b><c>#string</c></b></root>");
     }
 
+    @Test
+    void testXmlEqualsSchema(){
+        match("<root><a>x</a></root>", EQUALS_SCHEMA, "<root><a>#string</a><b><c>#string</c></b></root>");
+        match("<root><a>x</a><b></b></root>", EQUALS_SCHEMA, "<root><a>#string</a><b><c>#string</c></b></root>", FAILS);
+        match("<root><a>x</a><b><c></c></b></root>", EQUALS_SCHEMA, "<root><a>#string</a><b><c>#string</c></b></root>", FAILS);
+        match("<root><a>x</a><b><c>y</c></b></root>", EQUALS_SCHEMA, "<root><a>#string</a><b><c>#string</c></b></root>");
+    }
+
+    @Test
+    void testJsonEqualsSchema(){
+        match("{ a: 'x' }", EQUALS_SCHEMA, "{ a: '#string', b: { c: '#string' }}");
+        match("{ a: 'x', b:{} }", EQUALS_SCHEMA, "{ a: '#string', b: { c: '#string' }}");
+        match("{ a: 'x', b:{ c: 'y' } }", EQUALS_SCHEMA, "{ a: '#string', b: { c: '#string' }}");
+    }
```

Comment:
this seems to be a bug so I'm going to look at it, it should be a fix without needing any `SCHEMA` stuff

Reply:
actually no the second one is again not valid. here is a commit where I added the test that do work: https://github.com/intuit/karate/commit/98dd05e98c5c8730682b21ad20dc101dfaab1378